db/migrate/20210803171306_create_saved_searches_tables.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class CreateSavedSearchesTables < ActiveRecord::Migration[6.1]
+  def change
+    create_table :saved_searches do |t|
+      t.integer :user_id, null: false
+      t.string :query, null: false
+      t.string :compiled_query
+      t.integer :frequency, null: false
+      t.integer :last_post_id, null: false
+      t.datetime :last_searched_at, null: false
+      t.timestamps
+
+      t.index :user_id
+    end
+
+    execute <<~SQL
+      INSERT INTO saved_searches(user_id, query, frequency, last_post_id, last_searched_at, created_at, updated_at)
+      SELECT ucf1.user_id,
+             json_array_elements_text(ucf1.value::json->'searches'),
+             2,
+             COALESCE(ucf2.value::integer, 0),
+             ucf2.updated_at,
+             ucf1.created_at,
+             ucf1.updated_at
+      FROM user_custom_fields ucf1
+      LEFT JOIN user_custom_fields ucf2 ON ucf1.user_id = ucf2.user_id
+      WHERE ucf1.name = 'saved_searches' AND
+            ucf2.name = 'saved_searches_min_post_id'
```

Comment:
Did this migration fail for you? I explicitly used `LEFT JOIN` and then `COALESCE(ucf2.value::integer, 0)`.

There is a different bug however because I did not do the same for `last_searched_at` field.

Reply:
The migration 'succeeded', but none of my searches were migrated

Reply:
Ooh, got that one and fixed it. :)